clash-cores/src/Clash/Cores/Xilinx/Ethernet/Gmii/Types.hs

```diff
@@ -0,0 +1,160 @@
+-- | Contains the types used in the implementation of the GMII to SGMII bridge wrapper.
+{-# OPTIONS_GHC -fconstraint-solver-iterations=10 #-}
+{-# LANGUAGE RecordWildCards #-}
+module Clash.Cores.Xilinx.Ethernet.Gmii.Types where
```

Comment:
```suggestion
-- | Contains the types used in the implementation of the GMII to SGMII bridge wrapper.

{-# LANGUAGE RecordWildCards #-}

{-# OPTIONS_GHC -fconstraint-solver-iterations=10 #-}

module Clash.Cores.Xilinx.Ethernet.Gmii.Types where
```

Reply:
You really need 10 iterations? GHC 9.6.4 seems to need a mere godly 7.

I do realise we are engineers, but that doesn't mean always only using orders of magnitude :-D

Reply:
I like to keep a bit more overhead, otherwise I might have to bump to 8 on the next change :)

Reply:
It's not what I would do, but if that's how you want to do it I'm fine with it!

Reply:
On the 7 vs 10: the limit is just there to prevent the type checkers from going into an infinite loop. It doesn't actually cause more work. IMO the default should be such that users basically never see it.